parlai/agents/example_seq2seq/example_seq2seq.py

```diff
@@ -150,107 +145,82 @@ def share(self):
 
         return shared
 
-    def predict(self, xs, ys=None, is_training=False):
-        """Produce a prediction from our model.
-
-        Update the model using the targets if available.
-        """
+    def v2t(self, vector):
+        """Converts vector to text."""
+        if vector.dim() == 1:
+            output_tokens = []
+            # Remove the final END_TOKEN that is appended to predictions
+            for i, token in enumerate(vector):
+                if token == self.END_IDX:
+                    break
+                else:
+                    output_tokens.append(token)
+            return [self.dict.vec2txt(output_tokens)]
```

Comment:
It feels weird that (1d input) -> list[str] and (2d input) -> list[str]. I feel like you can always see it as a reduction of the last axis.

Illustration of the usage of [0] is where it particularly feels strange.

Reply:
sure that makes sense, adjusting